[17.0][MIG] intrastat_delivery: Migration to 17.0

nguyenminhchien opened the pull request:
Depends on intrastat_product

Commit:
[ADD] intrastat_product

Rename report.intrastat.product to l10n.fr.report.intrastat.product (same for service and for lines) + migration scripts

Rename field country_id to origin_country_id (TODO : write mig script) and move this field from l10n_fr_intrastat_product to intrastat_product

Move intrastat_code and uom_id of report.intrastat.code from l10n_fr_intrastat_product to intrastat_product, as discussed with Luc de Meyer

Use display_name instead of an inherit of name_get (better use of new API) Inherit write and create of intrastat codes to remove spaces (patch by Luc De Meyer from Noviat) Add O2M fields from intrastat code to products and product categs

Commit:
display_name must be stored as it is used as _rec_name, otherwise we cant, for instance, import files with name_search

Commit:
[IMP] intrastat_product: Several things:

* Update README.rst and translation files
* Add index=True on some fields on which we often search, as suggested by Yannick
* Remove description in __openerp__.py
* Move ACL from l10n_fr_intrastat_product module to intrastat_product module
* Large code re-engineering following the Akretion-Noviat code sprint on intrastat
  This code has been written both by Luc de Meyer and myself.
* Add first draft of code to generate decl lines

luc-demeyer committed:
[IMP] intrastat_product: major update intrastat V3 modules

* add region to intrastat_product

alexis-via committed:
[IMP+FIX] intrastat_product: Several things:

* Add product_origin_country_id on declaration/computation lines

  Copy incoterms and destination country from SO to invoice when invoicing from SO
  We need weight even when supplementary units is used
  Small cleanups and enhancements

* Add support for accessory costs

  Add default values for intrastat transaction on company
  Code cleanup

* If rounded weight is 0, put 1

  Take into account the taxes for B2C
  Small code cleanup

* Remove field exclude_from_intrastat

  Re-organise view of intrastat.product.declaration

* Add option intrastat_accessory_costs on company

  Set more fields as invisible (localisation should put them visible if they need it)
  Fix handling of suppl. units when hs_code is empty on invoice line (but set on product)
  Small usability enhancements

* Include selection of type of invoice in search, for better perf

  Isolate domain in a _prepare method, for easier inheritance
  WARNING: I changed the default selection of invoice type ; adapt it in your localization m
  odule if necessary
  Add intrastat_transaction_in_refund for company (not needed for France, but may be needed elsewhere)
  Add a log when an invoice line is skipped

* Include product code in warning msg on weight

* Inspired by the PR https://github.com/akretion/account-financial-reporting/pull/8 of Luc de Meyer

luc-demeyer committed:
intrastat fixes

alexis-via committed:
[FIX+IMP] intrastat_product: Fixes and imps:

* Improve logs and messages
* total_amount is a sum of integers, so it should be an integer
* Add transport mode in computation tree view
* FIX intrastat_country for invoices without src_dest_country_id
* FIX wrong model for seach method
* Use stock_picking_invoice_link for a better identification of the intrastat region

  With this commit, we now support the following scenario: I order to my supplier a quantity of 50 kg and he delivers/invoices 52kg ; odoo will create an additional invoice line of 2kg which is linked to the stock move, but not to any PO line.
* Modularise a piece of code
* Add ACL on hs.code to financial manager (I can't do it in product_harmonized_system because it doesn't depend on account)
* Handle scenario where an invoice has products with 0 value (samples for example) and shipping costs (accessory costs) with value > 0.

hbrunn committed:
[FIX] intrastat_product: this week's ci errors + another wrong variable name

alexis-via committed:
[FIX] intrastat_product: error message + Remove required=1

Because we can have intrastat unit that will never be mapped to an odoo unit (will always be set manually)

luc-demeyer committed:
update intrastat_product

alexis-via committed:
[MIG] intrastat_product: Migration to 10.0

intrastat_base: Move company view params to account config page

cubells committed:
[FIX] intrastat_product: state can be None

Commit:
res_company: fix intrastat_dispatches selection function

Commit:
[MIG] intrastat_product: Migration to 11.0

luc-demeyer committed:
[FIX+IMP] intrastat_product: reporting level + move methods to IntrastatProductDeclaration class

alexis-via committed:
|FIX] Broken intrastat decl form view (was borken when you did NOT have web_sheet_full_width)

luc-demeyer committed:
[10.0][FIX]get_hs_code_recursively on product.template

alexis-via committed:
Add new module product_harmonized_system_delivery (hide native hs_code field)

Finalize port to v11

alexis-via committed:
[FIX+IMP] intrastat_*: Several things:

* Add readonly on some fields when state of declaration is done
* Division by zero in computation of accessory costs
* Add FR translation
* Fix strings
* Minor code updates

luc-demeyer committed:
[MIG] intrastat_product: Migration to 11.0

* Small fixes in intrastat_base
* account_tax_template

pedrobaeza committed:
[FIX] intrastat_product: Fix duplicated msgid string

alexis-via committed:
Prepare v12 branch

alexis-via committed:
[MIG] Migrate all modules from v11 to v12

alexis-via committed:
[MIG] Migrate all modules from v11 to v12

luc-demeyer committed:
flake8

danielduqma committed:
[FIX] Translations in intrastat_product

oca-travis committed:
[UPD] Update product_harmonized_system_stock.pot

luc-demeyer committed:
[12.0][FIX]fix name_get on intrastat.transaction (#73)

sbidoul committed:
[FIX] intrastat_product USAGE.rst title level

OCA-git-bot committed:
[UPD] README.rst

Commit:
[MIG] product_harmonized_system_stock: Migration to 13.0

Commit:
[REF] product_harmonized_system_stock: Black python code

oca-travis committed:
[UPD] Update product_harmonized_system_stock.pot

OCA-git-bot committed:
[UPD] README.rst

luc-demeyer committed:
[MIG] intrastat_product: Migration to 13.0

BenjaHe committed:
[13.0][FIX] 'number' for invoice does not exist anymore the account.move has attribute 'name'

OCA-git-bot committed:
pre-commit update

luc-demeyer committed:
[13.0][FIX]invoice line hs code

alexis-via committed:
intrastat_product: Add xml_attachment_id field

intrastat_product: add mail.activity.mixin and widget in form view
product_harmonized_system: remove archive button (use Action menu)
All modules:
- use super() instead of super(class, self)
- Update domain_force in multi-company ir.rule to v13 standard

luc-demeyer committed:
[13.0]fix stack trace when entering invoice line without product

luc-demeyer committed:
refactor code to allow adding transaction details while encoding invoices

alexis-via committed:
[MIG] product_harmonized_system_stock to v14

alexis-via committed:
Update maintainers key

MiquelRForgeFlow committed:
[FIX] Migration fixes + full migration scripts

luc-demeyer committed:
[13.0]invoice_date -> date

joao-p-marques committed:
[MIG] intrastat_product: Migration to 14.0

Rename field type to declaration_type
Use the new intrastat boolean of fiscal position, following issue #110

Co-authored-by: Alexis de Lattre <alexis.delattre@akretion.com>

joao-p-marques committed:
[IMP] intrastat_product: add support for Brexit + small performance improvement + remove universal hs_code constaint

Add support for post-Brexit VAT numbers with latest python-stdnum version
Remove use of country_id.intrastat
Add 'vat' field on computation and declaration lines with a simple validity check
Add 'partner_id' field on computation lines
Prevent back to draft when an XML export is present
Remove the universal constaint applied to the hs_codes for countries in the "Europe" list, as discussed in https://github.com/OCA/intrastat-extrastat/pull/116#discussion_r569735555

Co-authored-by: Alexis de Lattre <alexis.delattre@akretion.com>

oca-travis committed:
[UPD] Update intrastat_product.pot

OCA-git-bot committed:
[UPD] README.rst

joao-p-marques committed:
[FIX] intrastat_product: Missed changes in migration

Change back from invoice.type to invoice.move_type
Update method signature and calls after the refactor in "intrastat.product.declaration"
Add some docs to explain the functionality that leads to this part of the code.

TT28080

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
intrastat_product 14.0.1.0.1

OCA-git-bot committed:
[UPD] README.rst

joao-p-marques committed:
[FIX] intrastat_product: don't force wrong domain in moves

The intrastat transaction domain is already forced in the model and in the field in account.move.
It was being incorrectly forced in the move, where it needed to be associated to the same company as the record, but those types can have no company associated

TT29582

oca-travis committed:
[UPD] Update product_harmonized_system_stock.pot

OCA-git-bot committed:
intrastat_product 14.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-14.0/intrastat-extrastat-14.0-product_harmonized_system_stock
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-14-0/intrastat-extrastat-14-0-product_harmonized_system_stock/

alexis-via committed:
intrastat_product: improve tree view of declaration

Commit:
Translated using Weblate (French)

Currently translated at 73.1% (177 of 242 strings)

Translation: intrastat-extrastat-14.0/intrastat-extrastat-14.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-14-0/intrastat-extrastat-14-0-intrastat_product/fr/

Commit:
Translated using Weblate (French)

Currently translated at 87.1% (211 of 242 strings)

Translation: intrastat-extrastat-14.0/intrastat-extrastat-14.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-14-0/intrastat-extrastat-14-0-intrastat_product/fr/

luc-demeyer committed:
[13.0][FIX]intrastat_product_declaration - fix amount by removing legacy currency conversion

victoralmau committed:
[IMP] intrastat_product: Add active field to transactions.
TT30682

oca-travis committed:
[UPD] Update intrastat_product.pot

OCA-git-bot committed:
intrastat_product 14.0.1.2.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-14.0/intrastat-extrastat-14.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-14-0/intrastat-extrastat-14-0-intrastat_product/

OCA-git-bot committed:
intrastat_product 14.0.1.3.0

rousseldenis committed:
[14.0][IMP] intrastat_product: Improve lines display

rousseldenis committed:
[14.0][IMP] intrastat_*: Improve tests and code

rousseldenis committed:
[14.0][IMP] intrastat_*: CONTRIBUTORS.rst

rousseldenis committed:
[14.0][IMP] intrastat_*: Add tests for base and for sale flows

rousseldenis committed:
[14.0][IMP] intrastat_product: Add test for declaration line values

rousseldenis committed:
[14.0][IMP] intrastat_product: Add test for purchase lines

rousseldenis committed:
[14.0][IMP] intrastat_product: Test declaration copy

rousseldenis committed:
[14.0][IMP] intrastat_product: Add hook for test values

OCA-git-bot committed:
intrastat_product 14.0.1.4.0

luc-demeyer committed:
cherry-pick of https://github.com/OCA/intrastat-extrastat/pull/151

OCA-git-bot committed:
intrastat_product 14.0.1.4.1

rousseldenis committed:
[IMP] intrastat_product: Better tests data

rousseldenis committed:
[14.0][IMP] intrastat_product: Add product country of origin in declaration

alexis-via committed:
intrastat_base: Remove intrastat.common class

rousseldenis committed:
[IMP] intrastat_product: Fix pre-commit

luc-demeyer committed:
[14.0]Intrastat Brexit support

luc-demeyer committed:
[14.0][FIX]intrastat_product_declaration - fix amount by removing legacy currency conversion

oca-travis committed:
[UPD] Update intrastat_product.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
intrastat_product 14.0.1.5.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-14.0/intrastat-extrastat-14.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-14-0/intrastat-extrastat-14-0-intrastat_product/

luc-demeyer committed:
[14.0]intrastat_product - fix amount_company_currency

luc-demeyer committed:
[14.0]intrastat_product - Intrastat Brexit support

OCA-git-bot committed:
intrastat_product 14.0.1.5.1

OCA-git-bot committed:
intrastat_product 14.0.1.5.2

luc-demeyer committed:
[14.0] brexit support - fix pipeline

luc-demeyer committed:
[14.0] brexit support - fix pipeline

rousseldenis committed:
[14.0][IMP] intrastat_product: Add tests for Excel generation

rousseldenis committed:
Updated from template

rousseldenis committed:
[14.0][IMP] intrastat_product: Add codes in Excel report as required in some countries

In some countries, the imports are done through Excel imports.

They need the codes only in the columns.

rousseldenis committed:
[14.0][IMP] intrastat_product: Add read access to intrastat lines to accounting base user

rousseldenis committed:
[IMP] intrastat_product: Add read accesses to intrastat lines

As Sale and Purchase users have access to invoices, we need to allow them
to read intrastat move lines as present on invoices.

luc-demeyer committed:
fix unit tests - keep product_origin_country_id on top of new product_origin_country_code to avoid breaking the localisation modules

luc-demeyer committed:
[14.0]intrastat_product - remove incorrect domain on comutation line

OCA-git-bot committed:
intrastat_product 14.0.1.5.3

luc-demeyer committed:
[14.0]intrastat - improved brexit support

luc-demeyer committed:
[14.0]improved brexit support - keep backwards compatibility with localisation modules

luc-demeyer committed:
[14.0]brexit support - increase test coverage

OCA-git-bot committed:
intrastat_product 14.0.1.5.4

luc-demeyer committed:
[14.0]brexit support - DeprecationWarning

luc-demeyer committed:
[14.0]brexit support - size=2 on product_origin_country_code

Commit:
[UPD] Update intrastat_product.pot

OCA-git-bot committed:
intrastat_product 14.0.1.6.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-14.0/intrastat-extrastat-14.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-14-0/intrastat-extrastat-14-0-intrastat_product/

luc-demeyer committed:
[14.0]fix stack trace when running module upgrade on 14.0.1.6.0

Commit:
[UPD] Update intrastat_product.pot

OCA-git-bot committed:
intrastat_product 14.0.1.6.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-14.0/intrastat-extrastat-14.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-14-0/intrastat-extrastat-14-0-intrastat_product/

victoralmau committed:
Added translation using Weblate (Spanish)

victoralmau committed:
Translated using Weblate (Spanish)

Currently translated at 0.7% (2 of 261 strings)

Translation: intrastat-extrastat-14.0/intrastat-extrastat-14.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-14-0/intrastat-extrastat-14-0-intrastat_product/es/

victoralmau committed:
[IMP] intrastat_product: Add an Intrastat summary in the invoice report.
TT37168

StefanRijnhart committed:
[IMP] intrastat_product: prepopulate stored computed fields

StefanRijnhart committed:
[FIX] intrastat_product: typo in filename

OCA-git-bot committed:
intrastat_product 14.0.1.6.2

Commit:
[UPD] Update intrastat_product.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
intrastat_product 14.0.2.0.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-14.0/intrastat-extrastat-14.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-14-0/intrastat-extrastat-14-0-intrastat_product/

victoralmau committed:
[IMP] intrastat_product: Skip invoices according to partner country

Accidentally removed in https://github.com/OCA/intrastat-extrastat/commit/2646dc489ed76000d5eb0356ded1a71d60bf5a67

TT38922

OCA-git-bot committed:
intrastat_product 14.0.2.0.1

Commit:
[UPD] Update intrastat_product.pot

OCA-git-bot committed:
intrastat_product 14.0.2.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-14.0/intrastat-extrastat-14.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-14-0/intrastat-extrastat-14-0-intrastat_product/

luc-demeyer committed:
intrastat declaration - fix excel export

luc-demeyer committed:
intrastat declaration - restore _compute_numbers that was dropped by PR #149 (removal of intrastat.common

alexis-via committed:
[IMP] Big simplification, update and cleanup

This commit aims at removing the over-complexity of intrastat modules
while increasing simplicity/usability for users.

- Move default intrastat transaction from res.company to account.fiscal.position to add su
pport for B2C (and not just B2B)
- improve usability: auto-generate declaration lines and XML export when
going from draft to done. Auto-delete declaration lines and XML export
when going from done to draft (and add confirmation pop-up).
- declaration lines are now readonly. Only computation lines can be created/edited manuall
y
- add field region_code on computation lines and declaration lines.
Remove region_id on declaration lines. This change allows big
simplification in some localization modules such as
l10n_fr_intrastat_product.
- simplify Brexit implementation. Northern Ireland is important, but we
can't afford to have so many lines of code and add a field on
product.template (origin_state_id) for a territory of 1.9 million
inhabitants! This is too costly to maintain and too complex for users.
- improve default visibility of fields when reporting_level = 'standard'
- add support for weight calculation from uom categories other than
units and weight, supposing that the 'weight' field on product.template
is the weight per uom of the product
- add EU companies from several different countries in demo data with valid VAT numbers

OCA-git-bot committed:
intrastat_product 14.0.2.1.1

OCA-git-bot committed:
intrastat_product 14.0.2.1.2

alexis-via committed:
[MIG] product_harmonized_system_stock to v15

Commit:
[UPD] Update product_harmonized_system_stock.pot

OCA-git-bot committed:
[UPD] README.rst

alexis-via committed:
[MIG] product_harmonized_system_stock to v16

victoralmau committed:
[FIX] intrastat_product: Avoid error if we create lines manually

TT40919

OCA-git-bot committed:
intrastat_product 14.0.2.1.3

alexis-via committed:
intrastat_product: default intrastat transaction on fiscal position AND company

Odoo uses, by order of priority:
1) intrastat transaction of the invoice
2) default intrastat transaction of the fiscal position of the invoice
3) default intrastat transaction of the company

alexis-via committed:
intrastat_product: move res_config_settings.py to wizard subdir

alexis-via committed:
intrastat_product: Improve warning message on intrastat transaction

alexis-via committed:
intrastat_product: re-write the generation of the messages in the warning pop-up

This is completely new implementation of the messages that are displayed
in the warning pop-up:
- no double messages
- messages are grouped by section and record
- nice HTML display insead of raw text

alexis-via committed:
intrastat_product: Backport updates from v15

Make tests work

alexis-via committed:
[MIG] intrastat_product to v16

This migration has been made from the v14 branch akretion/14-brexit-simplification-and-type-on-fp and the changes of v15 has been re-integrated.

As discussed on https://github.com/OCA/intrastat-extrastat/issues/194
the intrastat transaction codes are now integrated in the module intrastat_product

The default value for transaction_id depending on the move_type of the
invoice have been dropped on fiscal position and company: Odoo sets 11
when fiscal position is B2B and 12 when fiscal position is B2C.
As a consequence: intrastat.transaction: company_id = False by default

Use textwrap for name_get() of intrastat.transaction

partner_id on computation line is not a related field any more: it allows
to auto-fill VAT from partner_id when creating a line manually.

computation and declaration line tree view: display transaction code
instead of transaction_id to save space

alexis-via committed:
intrastat_product: single XLSX export

There is now a single XLSX export that has 2 tabs:
1) computation lines
2) declaration lines
The declaration type is now mentionned in the XLSX export.
Ability to export several declarations in 1 XLSX file.

Commit:
[UPD] Update product_harmonized_system_delivery.pot

OCA-git-bot committed:
[UPD] README.rst

Commit:
[UPD] Update product_harmonized_system_stock.pot

OCA-git-bot committed:
[UPD] README.rst

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (5 of 5 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system_delivery
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system_delivery/hr/

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (1 of 1 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system_stock
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system_stock/hr/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system_delivery
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system_delivery/

Gelo-fl committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (3 of 3 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system_delivery
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system_delivery/es/

sanrav committed:
Added translation using Weblate (French (France) (fr_FR))

sanrav committed:
Translated using Weblate (French (France) (fr_FR))

Currently translated at 100.0% (1 of 1 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system_stock
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system_stock/fr_FR/

Commit:
Added translation using Weblate (French)

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (5 of 5 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system_delivery
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system_delivery/fr/

Commit:
Added translation using Weblate (French)

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (1 of 1 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system_stock
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system_stock/fr/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system_delivery
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system_delivery/

Commit:
[UPD] Update intrastat_product.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-intrastat_product/

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (258 of 258 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-intrastat_product/fr/

alexis-via committed:
intrastat_product: fix strings

Commit:
[UPD] Update intrastat_product.pot

OCA-git-bot committed:
intrastat_product 16.0.1.1.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-intrastat_product/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (256 of 256 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-intrastat_product/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (5 of 5 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system_delivery
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system_delivery/es/

metaminux committed:
[FIX] intrastat_product : properly initialize notedict before calling _get_weight_and_supplunits
  fixes OCA/intrastat-extrastat#224

metaminux committed:
[FIX] intrastat_product : uom_id in intrastat unit form view shouldn't be required
  Fixes OCA/intrastat-extrastat#223

Commit:
[UPD] Update intrastat_product.pot

OCA-git-bot committed:
intrastat_product 16.0.1.2.0

OCA-git-bot committed:
intrastat_product 16.0.1.2.1

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-intrastat_product/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (256 of 256 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-intrastat_product/es/

Commit:
Added translation using Weblate (French (Belgium))

Commit:
Translated using Weblate (French)

Currently translated at 97.2% (249 of 256 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-intrastat_product/fr/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (256 of 256 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-intrastat_product/es/

alexis-via committed:
intrastat_product: use shipping partner instead of invoicing partner if any

Replace @api.onchange by compute methods with store=True, precompute=True and readonly=False for vat and region_code fields of computation lines

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-intrastat_product/

victoralmau committed:
[FIX] intrastat_product: Prevent error when adding transaction details on invoices

The exclude_from_invoice_tab field does not exist in v16

Commit:
[UPD] Update intrastat_product.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-intrastat_product/

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[IMP] intrastat_product: Set translate=True to description field from transactions

In v16 the transactions in data have been added, it will now be necessary
that the localization modules do not add new ones, but it is also convenient
to display the corresponding text in each language.

Commit:
[UPD] Update intrastat_product.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-intrastat_product/

OCA-git-bot committed:
[BOT] post-merge updates

alexis-via committed:
intrastat_product: suppl_unit_qty rounded to 1 if between 0 and 0.5

Improve comment in the code about rounding

jdidderen committed:
[IMP] intrastat_product - remove default company for intrastat regions

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

jdidderen committed:
[FIX] intrastat_product - set region_code based on code or the region

OCA-git-bot committed:
[BOT] post-merge updates

alexis-via committed:
[IMP] intrastat_product: method for XLSX filename

aritzolea committed:
[FIX] intrastat_product: Display note field as html on product declatation

OCA-git-bot committed:
[BOT] post-merge updates

aritzolea committed:
[FIX] intrastat_product: Do not display missing VAT number error when fiscal position is B2C

Commit:
[UPD] Update intrastat_product.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-intrastat_product/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (274 of 274 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-intrastat_product/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (274 of 274 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-intrastat_product/es/

OCA-git-bot committed:
[BOT] post-merge updates

aritzolea committed:
[ADD] intrastat_delivery

Commit:
[UPD] Update intrastat_delivery.pot

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (5 of 5 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-intrastat_delivery
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-intrastat_delivery/es/

alexis-via committed:
intrastat_product: add confirmed state

Add a third step "confirmed" between "draft" and "done". It is necessary
to have error messages of XML file generation that refer to declaration
lines.

Add line_number on declaration line, to be used in error message when
generating XML file.

note field moved from fields.Text to fields.Html

Improve display_name of intrastat.product.declaration

Add store=True on 3 computed fields (reporting_level on
declaration, src_dest_country_code and product_origin_country_code on
computation lines). It fixes a bug on src_dest_country_code and
product_origin_country_code on computation lines, when you could not set
manually a country code while leaving the m2o field empty.

alexis-via committed:
intrastat_product: add readonly ACL for auditor

alexis-via committed:
intrastat_product: auto reformatting to prepare v17 migration

alexis-via committed:
[MIG] intrastat_product to v17

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system_stock
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system_stock/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system_delivery
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system_delivery/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-intrastat_delivery
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-intrastat_delivery/it/

AndreuOForgeFlow committed:
[IMP] product_harmonized_system_stock: pre-commit auto fixes

AndreuOForgeFlow committed:
[MIG] product_harmonized_system_stock: Migration to 17.0

AndreuOForgeFlow committed:
[IMP] product_harmonized_system_delivery: pre-commit auto fixes

AndreuOForgeFlow committed:
[MIG] product_harmonized_system_delivery: Migration to 17.0

nguyenminhchien committed:
[IMP] intrastat_delivery: pre-commit auto fixes

dzungtran89 commented on intrastat_delivery/models/sale_order.py:
```diff
@@ -8,12 +8,15 @@ class SaleOrder(models.Model):
-                }
-            )
+            vals = {}
+            if order.carrier_id.incoterm:
```
@nguyenminhchien this would be incorrect in the case:
- at time `t`, user confirms the SO, which the carrier has defined incoterm
- at time `t+1`, user removes the incoterm on carrier, set SO to draft, and re-confirm it
  The SO will keep the previous incoterm on the SO

nguyenminhchien replied:
i got it, but if keep as before: always update `vals["incoterm"] = order.carrier_id.incoterm.id`. it does not respect what written in the README:

![image](https://github.com/nguyenminhchien/intrastat-extrastat/assets/8724793/e40abe46-45e0-4ffa-836b-2a8604eccb82)

Do you have any suggestion here?

dzungtran89 replied:
nice catch, if so, that will be a fix in a seperated commit

nguyenminhchien replied:
ok, i have rollback these lines.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (22 of 22 strings)

Translation: intrastat-extrastat-17.0/intrastat-extrastat-17.0-intrastat_base
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-17-0/intrastat-extrastat-17-0-intrastat_base/it/

OCA-git-bot committed:
Merge PR #269 into 17.0

Signed-off-by gurneyalex

Commit:
[UPD] Update product_harmonized_system_stock.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #270 into 17.0

Signed-off-by gurneyalex

Commit:
[UPD] Update product_harmonized_system_delivery.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-17.0/intrastat-extrastat-17.0-product_harmonized_system_delivery
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-17-0/intrastat-extrastat-17-0-product_harmonized_system_delivery/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: intrastat-extrastat-17.0/intrastat-extrastat-17.0-product_harmonized_system_stock
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-17-0/intrastat-extrastat-17-0-product_harmonized_system_stock/it/

OCA-git-bot committed:
[IMP] update dotfiles

OCA-git-bot committed:
[IMP] update dotfiles

dreispt committed:
[REF] intrastat_product: fix lint

dreispt committed:
[FIX] intrastat_product: error on Transaction Type name computation

OCA-git-bot committed:
Merge PR #276 into 17.0

Signed-off-by gurneyalex

Commit:
[UPD] Update intrastat_product.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-17.0/intrastat-extrastat-17.0-intrastat_product
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-17-0/intrastat-extrastat-17-0-intrastat_product/

nguyenminhchien committed:
[MIG] intrastat_delivery: Migration to 17.0
